lib/db/mysql.js

```diff
@@ -283,8 +283,52 @@ module.exports = function (log, error) {
         deviceInfo.name,
         deviceInfo.type,
         deviceInfo.createdAt,
-        deviceInfo.callbackURL
-      ]
+        deviceInfo.callbackURL,
+        deviceInfo.callbackPublicKey
+      ],
+      null,
+      function (err) {
+        if (err.errno === ER_DUP_ENTRY) {
+          if (err.toString().match(/ for key 'UQ_devices_sessionTokenId'$/)) {
```

Comment:
This is pretty flakey, I realise, but it's the only way I could find to distinguish the two errors. But maybe distingushing between them is unnecessary and it's fine if they both return the same `errno`? I'd *love* to get rid of this if that is the case. :smile: